### PR TITLE
Fix #2142: "Continue" Button increased its top/bottom padding

### LINKED ISSUE #2142: "Continue" Button increased its top/bottom padding
There are several instances where the top/bottom padding of the continue button increases. After tapping "Continue" on one page, the "Continue" button on the following page appears taller. [[Video](https://drive.google.com/file/d/1g4kJB4UuD4D-ZpgA64EAqUMrmdmg1hbQ/view?usp=sharing)]
The video is recorded in topic **Place Values**, story **Jaime's Adventures in the Arcade**, **Chapter 1: What are the place values**?

### PR DESCRIPTION
**Pull request by 7phalange7**
## Explanation

**Fixes #2142**
### Before :&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;      After :
before_ss&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;after_ss

after_ss&nbsp;after_ss

 
 

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Review by rt4914 — COMMENTED** [review]
@7phalange7 Please make similar changes for Tablet layout too.

**Comment by 7phalange7** [comment]
> 
> 
> @7phalange7 Please make similar changes for Tablet layout too.

okay, I will do that. Sorry, I overlooked it by mistake.

File: app/src/main/res/layout-land/continue_interaction_item.xml [hunk 1]
```diff
@@ -45,9 +45,9 @@
       android:onClick="@{(v) -> viewModel.handleButtonClicked()}"
       android:text="@string/state_continue_button"
       android:textColor="@color/white"
-      app:layout_constraintBottom_toBottomOf="parent"
+      app:layout_constraintBottom_toBottomOf="@+id/previous_state_navigation_button"
       app:layout_constraintEnd_toEndOf="parent"
       app:layout_constraintHeight_min="48dp"
-      app:layout_constraintTop_toTopOf="parent" />
+      app:layout_constraintTop_toTopOf="@+id/previous_state_navigation_button" />
```
Comment (anandwana001) [inline app/src/main/res/layout-land/continue_interaction_item.xml]: `@id/previous_state_navigation_button`
ditto for others

Comment (anandwana001) [reply app/src/main/res/layout-land/continue_interaction_item.xml]: `@+id/previous_state_navigation_button` means creating an id in the namespace. 

`@id/previous_state_navigation_button` means referencing the id which is already defined in the android namespace. 

I don't know why but this is something which Android Studio needs to catch.

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thanks @7phalange7 
nit change suggested

File: app/src/main/res/layout-land/continue_interaction_item.xml (hunk 1 above)
Comment (7phalange7) [reply app/src/main/res/layout-land/continue_interaction_item.xml]: Okay. I wasn't aware of the difference b/w `id` and `+id` . I'll make the required changes asap. Thank you

**Comment by 7phalange7** [comment]
I went through the code and test logs. I am not able to determine why the test failed. Please guide me. @anandwana001

**Comment by BenHenning** [comment]
@7phalange7 which test is failing for you? It seems that all tests are passing on GitHub Actions. If you could provide more context about what failure you're seeing (e.g. logs) + the steps you took to hit that case, we will be able to provide better help. Also, if your local changes are failing but not the ones on the PR, I suggest pushing your changes so that we can see what code is leading to the failure.

**Comment by 7phalange7** [comment]
@BenHenning previously Github Actions was showing a failing test, but now it seems fine.

File: app/src/main/res/layout-land/continue_interaction_item.xml (hunk 1 above)
Comment (BenHenning) [reply app/src/main/res/layout-land/continue_interaction_item.xml]: Thanks @7phalange7. It seems like some of the new IDs aren't using ``@+id`` yet--did you make the requested changes in all of the new locations? Looking at https://github.com/oppia/oppia-android/pull/2197/files shows cases of just ``@id/...``.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @7phalange7! Overall, I think this looks more correct. However, I have one main concern besides the one that @anandwana001 already mentioned above: how does this UI change look when there is no previous button visible? Can you provide before/after screenshots for the case when the continue button is on the first card of an exploration (see prototype exploration)? I'm not actually sure how constraint layout handles constraining against views when they're gone, so we should verify this case.

**Comment by 7phalange7** [comment]
@BenHenning Yes you are right, I just noticed that when `previous` button is not visible , the `continue` button is pushed down. In hindsight directly fixing the height of `continue` button to 48dp would have been a better option and it does not depend upon the state of previous button. What do you think about fixing the height to 48dp as the previous button height is fixed already to 48dp?

### Before :&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;      After(current commit) : &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;       After(fixing button height to 48dp) :
before_ss&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;after_ss  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;after_ss
before_ss&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;after_ss  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;after_ss

**Comment by 7phalange7** [comment]
The previous solution causes the `continue` button to shift down when the `previous` button is `gone` (as shown in the above comment) so I'll be committing a new solution - hardcode the height of continue button to `48dp`

**Review by rt4914 — APPROVED** [review]
LGTM, will wait for @BenHenning thought on this one.

**Review by BenHenning — APPROVED** [review]
Thanks @7phalange7! I unfortunately forgot why we used minHeight before, but this seems fine to me. I suspect the constraints we were using originally when that was checked in might have been different--it seems more correct now.

**Comment by BenHenning** [comment]
@anandwana001 it seems you requested changaes. PTAL.

**Review by anandwana001 — APPROVED** [review]
LGTM

**Pull request merged**